testing/run_tests.py

```diff
@@ -0,0 +1,229 @@
+  kernel_file_output = os.path.join(out_dir, kernel_file_name)
+  
+  SnapshotTest(build_dir, dart_file, kernel_file_output)
+
+  print "Running test '%s' using 'flutter_tester'" % kernel_file_name
+  RunEngineExecutable(build_dir, 'flutter_tester', filter, [
+    '--disable-observatory',
+    '--use-test-fonts',
+    kernel_file_output
+  ])
+
+def RunPubGet(build_dir, directory):
+  print "Running 'pub get' in the tests directory %s" % dart_tests_dir
+
+  pub_get_command = [
+    os.path.join(build_dir, 'dart-sdk', 'bin', 'pub'),
+    'get'
+  ]
+  subprocess.check_call(pub_get_command, cwd=directory)
+
+
+def EnsureDebugUnoptSkyPackagesAreBuilt():
+  variant_out_dir = os.path.join(out_dir, 'host_debug_unopt')
+
+  ninja_command = [
+    'autoninja',
+    '-C',
+    variant_out_dir,
+    'flutter/sky/packages'
+  ]
+
+  # Attempt running Ninja if the out directory exists.
+  # We don't want to blow away any custom GN args the caller may have already set.
+  if os.path.exists(variant_out_dir):
+    subprocess.check_call(ninja_command, cwd=buildroot_dir)
+    return
+
+  gn_command = [
+    os.path.join(buildroot_dir, 'flutter', 'tools', 'gn'),
+    '--runtime-mode',
+    'debug',
+    '--unopt',
+    '--no-lto',
+  ]
+  
+  subprocess.check_call(gn_command, cwd=buildroot_dir)
+  subprocess.check_call(ninja_command, cwd=buildroot_dir)
+
+def RunDartTests(build_dir, filter):
+  # This one is a bit messy. The pubspec.yaml at flutter/testing/dart/pubspec.yaml
+  # has dependencies that are hardcoded to point to the sky packages at host_debug_unopt/
+  # Before running Dart tests, make sure to run just that target (NOT the whole engine)
+  EnsureDebugUnoptSkyPackagesAreBuilt();
+
+  # Now that we have the Sky packages at the hardcoded location, run `pub get`.
+  RunPubGet(build_dir, dart_tests_dir)
+
+  dart_tests = glob.glob('%s/*.dart' % dart_tests_dir)
+
+  for dart_test_file in dart_tests:
+    if filter in os.path.basename(dart_test_file):
+      print "Testing dart file %s" % dart_test_file
+      RunDartTest(build_dir, dart_test_file, filter)
+    else:
+      print "Skipping %s due to filter." % dart_test_file
+
+
+def RunTests(build_dir, filter, run_engine_tests, run_dart_tests, run_benchmarks):
+  if run_engine_tests:
+    RunEngineTests(build_dir, filter)
+
+  if run_dart_tests:
+    RunDartTests(build_dir, filter)
+
+  if run_benchmarks:
+    RunEngineBenchmarks(build_dir, filter)
+
+
+def main():
+  parser = argparse.ArgumentParser();
+
+  parser.add_argument('--variant', dest='variant', action='store', 
+      default='host_debug_unopt', help='The engine build variant to run the tests for.');
+  parser.add_argument('--type', type=str, choices=['all', 'engine', 'dart', 'benchmarks'], default='all')
+  parser.add_argument('--filter', type=str, default='',
+      help='The file name filter to use to select specific tests to run.')
+
+  args = parser.parse_args()
+  
+  run_engine_tests = args.type == 'engine'
```

Comment:
Nit: `args.type in ['engine', 'all']` to save 4 lines below :)

Reply:
Cool. Done.